httpbinding/user_agent.go

```diff
@@ -0,0 +1,38 @@
+package httpbinding
+
+import (
+	"fmt"
+	"strings"
+)
+
+// UserAgentBuilder is a builder for a HTTP User-Agent string.
+type UserAgentBuilder struct {
+	sb strings.Builder
+}
```

Comment:
Curious if user agent is only added for http based protocol. That seems incorrect, so probably not?

Reply:
HTTP is the only protocol we support, since we don't have a standard for other protocols and their user-agents it would be premature to make the utility generic.